crates/bevy_animation/src/animation_curves.rs

```diff
@@ -242,20 +259,72 @@ where
         self.curve.domain()
     }
 
-    fn apply<'a>(
-        &self,
-        t: f32,
-        _transform: Option<Mut<'a, Transform>>,
-        mut entity: AnimationEntityMut<'a>,
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
```

Comment:
I don't think this fn is necessary, since `Reflect` already has `as_any`, and `AnimatableCurve` is `Reflect`.
I tried removing this fn and replacing `AnimatableCurve::as_any` with `Reflect::as_any`, and the animation_graph example seems to work fine, unless I've missed something?

<details>
<summary>Patch</summary>

```diff
diff --git a/crates/bevy_animation/src/animation_curves.rs b/crates/bevy_animation/src/animation_curves.rs
index d8eb8241c..09aa96d14 100644
--- a/crates/bevy_animation/src/animation_curves.rs
+++ b/crates/bevy_animation/src/animation_curves.rs
@@ -78,7 +78,7 @@
 //! [there]: AnimatableProperty

 use core::{
-    any::{Any, TypeId},
+    any::TypeId,
     fmt::{self, Debug, Formatter},
     marker::PhantomData,
 };
@@ -259,10 +259,6 @@ where
         self.curve.domain()
     }

-    fn as_any(&self) -> &dyn Any {
-        self
-    }
-
     fn create_evaluator(&self) -> Box<dyn AnimationCurveEvaluator> {
         Box::new(AnimatableCurveEvaluator {
             evaluator: BasicAnimationCurveEvaluator::default(),
@@ -314,7 +310,7 @@ where
         weight: f32,
         graph_node: AnimationNodeIndex,
     ) -> Result<(), AnimationEvaluationError> {
-        let curve = (*AnimationCurve::as_any(curve))
+        let curve = (*Reflect::as_any(curve))
             .downcast_ref::<AnimatableCurve<P, C>>()
             .unwrap();
         let value = curve.curve.sample_clamped(t);
@@ -357,10 +353,6 @@ where
         Box::new(self.clone())
     }

-    fn as_any(&self) -> &dyn Any {
-        self
-    }
-
     fn domain(&self) -> Interval {
         self.0.domain()
     }
@@ -413,7 +405,7 @@ where
         weight: f32,
         graph_node: AnimationNodeIndex,
     ) -> Result<(), AnimationEvaluationError> {
-        let curve = (*AnimationCurve::as_any(curve))
+        let curve = (*Reflect::as_any(curve))
             .downcast_ref::<TranslationCurve<C>>()
             .unwrap();
         let value = curve.0.sample_clamped(t);
@@ -456,10 +448,6 @@ where
         Box::new(self.clone())
     }

-    fn as_any(&self) -> &dyn Any {
-        self
-    }
-
     fn domain(&self) -> Interval {
         self.0.domain()
     }
@@ -512,7 +500,7 @@ where
         weight: f32,
         graph_node: AnimationNodeIndex,
     ) -> Result<(), AnimationEvaluationError> {
-        let curve = (*AnimationCurve::as_any(curve))
+        let curve = (*Reflect::as_any(curve))
             .downcast_ref::<RotationCurve<C>>()
             .unwrap();
         let value = curve.0.sample_clamped(t);
@@ -555,10 +543,6 @@ where
         Box::new(self.clone())
     }

-    fn as_any(&self) -> &dyn Any {
-        self
-    }
-
     fn domain(&self) -> Interval {
         self.0.domain()
     }
@@ -611,7 +595,7 @@ where
         weight: f32,
         graph_node: AnimationNodeIndex,
     ) -> Result<(), AnimationEvaluationError> {
-        let curve: &ScaleCurve<C> = (*AnimationCurve::as_any(curve)).downcast_ref().unwrap();
+        let curve: &ScaleCurve<C> = (*Reflect::as_any(curve)).downcast_ref().unwrap();
         let value = curve.0.sample_clamped(t);
         self.evaluator
             .stack
@@ -655,10 +639,6 @@ where
         Box::new(self.clone())
     }

-    fn as_any(&self) -> &dyn Any {
-        self
-    }
-
     fn domain(&self) -> Interval {
         self.0.domain()
     }
@@ -711,7 +691,7 @@ where
         weight: f32,
         graph_node: AnimationNodeIndex,
     ) -> Result<(), AnimationEvaluationError> {
-        let curve: &TransformCurve<C> = (*AnimationCurve::as_any(curve)).downcast_ref().unwrap();
+        let curve: &TransformCurve<C> = (*Reflect::as_any(curve)).downcast_ref().unwrap();
         let value = curve.0.sample_clamped(t);
         self.evaluator
             .stack
@@ -784,10 +764,6 @@ where
         self.0.domain()
     }

-    fn as_any(&self) -> &dyn Any {
-        self
-    }
-
     fn create_evaluator(&self) -> Box<dyn AnimationCurveEvaluator> {
         Box::new(WeightsCurveEvaluator {
             stack_morph_target_weights: vec![],
@@ -889,7 +865,7 @@ where
         weight: f32,
         graph_node: AnimationNodeIndex,
     ) -> Result<(), AnimationEvaluationError> {
-        let curve: &WeightsCurve<C> = (*AnimationCurve::as_any(curve)).downcast_ref().unwrap();
+        let curve: &WeightsCurve<C> = (*Reflect::as_any(curve)).downcast_ref().unwrap();

         let prev_morph_target_weights_len = self.stack_morph_target_weights.len();
         self.stack_morph_target_weights
@@ -1003,9 +979,6 @@ pub trait AnimationCurve: Reflect + Debug + Send + Sync {
     /// Returns a boxed clone of this value.
     fn clone_value(&self) -> Box<dyn AnimationCurve>;

-    /// Returns this value upcast to [`Any`].
-    fn as_any(&self) -> &dyn Any;
-
     /// The range of times for which this animation is defined.
     fn domain(&self) -> Interval;
```
</details>

Reply:
I removed it.